GitUI/CommandsDialogs/FormResolveConflicts.cs

```diff
@@ -874,38 +894,115 @@ private void ChooseRemoteOnConflict(string fileName)
+            // Local
+            var btnKeepLocal = new TaskDialogCommandLink("KeepLocal", null, keepLocalButtonText);
+            btnKeepLocal.Click += (s, e) =>
             {
-                frm.ShowDialog(this);
+                _solveMergeConflictDialogResult = SolveMergeConflictActionType.atKeepLocal;
+                dialog.Close();
+            };
 
-                if (frm.KeepBase)
-                {
-                    // base
-                    ChooseBaseOnConflict(item.Filename);
-                }
+            // Remote
+            var btnKeepRemote = new TaskDialogCommandLink("KeepRemote", null, keepRemoteButtonText);
+            btnKeepRemote.Click += (s, e) =>
+            {
+                _solveMergeConflictDialogResult = SolveMergeConflictActionType.atKeepRemote;
+                dialog.Close();
+            };
 
-                if (frm.KeepLocal)
-                {
-                    // local
-                    ChooseLocalOnConflict(item.Filename);
-                }
+            // Base
+            var btnKeepBase = new TaskDialogCommandLink("KeepBase", null, keepBaseButtonText);
+            btnKeepBase.Click += (s, e) =>
+            {
+                _solveMergeConflictDialogResult = SolveMergeConflictActionType.atKeepBase;
+                dialog.Close();
+            };
+
+            dialog.Controls.Add(btnKeepLocal);
+            dialog.Controls.Add(btnKeepRemote);
+            dialog.Controls.Add(btnKeepBase);
+
+            return dialog;
+        }
+
+        private void OpenSolveMergeConflictDialogAndExecuteSelectedMergeAction(SolveMergeConflictCheckRevisionDelegate selectedMergeAction,
+            string dialogText, string dialogInstructionText, string dialogCaption, string dialogFooterCheckboxText,
+            string keepLocalButtonText, string keepRemoteButtonText, string keepBaseButtonText)
+        {
+            bool shouldOpenDialog = _SolveMergeConflictApplyToAll == false;
+            if (shouldOpenDialog)
+            {
+                using var dialog = CreateSolveMergeConflictTaskDialog(Handle, dialogText, dialogInstructionText, dialogCaption, dialogFooterCheckboxText,
+                    keepLocalButtonText, keepRemoteButtonText, keepBaseButtonText);
+
+                dialog.Show();
+                _SolveMergeConflictApplyToAll = dialog.FooterCheckBoxChecked ?? false;
+                _SolveMergeConflictLastAction = _solveMergeConflictDialogResult; // The value of _solveMergeConflictDialogResult is changed in the method CreateSolveMergeConflictTaskDialog(...)
+            }
+
+            selectedMergeAction(_solveMergeConflictDialogResult);
+        }
 
-                if (frm.KeepRemote)
+        private void BinaryFilesChooseLocalBaseRemote(ConflictData item)
+        {
+            SolveMergeConflictCheckRevisionDelegate executeAction = (dialogResult) =>
+            {
+                switch (dialogResult)
```

Comment:
Use pattern matching, it will make the code denser and easier to read

Reply:
Honestly speaking, I am not sure what you are meaning with that. I did change the code, maybe the change is somehow already there ;-) Please take a look, thank you!